dmoj/contrib/coci.py

```diff
@@ -1,26 +1,26 @@
-import re
-
-from dmoj.contrib.testlib import ContribModule as TestlibContribModule
-from dmoj.error import InternalError
-from dmoj.result import CheckerResult
-
-
-class ContribModule(TestlibContribModule):
-    name = 'coci'
-    repartial = re.compile(br'^partial ((\d+)\/(\d*[1-9]\d*))\n$')
-
-    @classmethod
-    def parse_return_code(cls, proc, executor, point_value, time_limit, memory_limit, feedback, name, stderr):
-        if proc.returncode == cls.PARTIAL:
-            match = cls.repartial.search(stderr)
-            if not match:
-                raise InternalError('Invalid stderr for fraction of points: %r' % stderr)
-            percentage = int(match.group(2)) / int(match.group(3))
-            if not 0.0 <= percentage <= 1.0:
-                raise InternalError('Invalid fraction: %s' % match.group(1))
-            points = percentage * point_value
-            return CheckerResult(True, points, feedback=feedback)
-        else:
-            return super().parse_return_code(
-                proc, executor, point_value, time_limit, memory_limit, feedback, name, stderr
-            )
+import re
+
+from dmoj.contrib.testlib import ContribModule as TestlibContribModule
+from dmoj.error import InternalError
+from dmoj.result import CheckerResult
+
+
+class ContribModule(TestlibContribModule):
+    name = 'coci'
+    repartial = re.compile(br'^partial ((\d+)\/(\d*[1-9]\d*))\n$', re.M)
```

Comment:
Do we need/wait the `\n`?

Reply:
IMO we should maintain parity with testlib.h, whatever that may be...

Also I would recommend against removing entirely, as then we'd have to modify every checker that uses partials...